### ISSUE TITLE
Reviews - edition 2, round 2, part 1

### ISSUE THREAD

**Issue by Nowosad** [geocompx/geocompr#898]
- [x] [jn] p. 2 -- Should "adversial" be "adversarial"? [Fixed]
- [x] [jm] p. 2 -- You say that GUi systems "provide access to spatial database management systems with object-oriented relational data models, topology, and fast (spatial) querying." I'm not sure how this access to databases would be unique to GUIs, whereas the other four bullet points in this list do seem unique to GUIs instead of command line systems. [Fair point. At the time of writing I was possibly thinking of ArcGIS's geodatabase and GRASS GIS's database. However, you are right, one can use these through the cli as well. Hence, I deleted this bullet point from the list. Thanks for noting.]
- [x] [jn] p. 3 -- If all GISs listed are hybrid, why include the Support column?. [It is just to highlight that all of the listed tools support, currently, both vector and raster operations. That was not always the case.]
- [x] [jn] p. 6 -- You write that "apparently, these are INPUT, OVERLAY, OVERLAY_FIELDS_PREFIX, and OUTPUT." These are, though, not evident from the book when the output is not presented. While the output is, as you note, "rather long", could a portion of it be presented to illustrate this? [We simplified the sentence to: "These arguments are `INPUT`, `OVERLAY`, `OVERLAY_FIELDS_PREFIX`, and `OUTPUT`."]
- [x] [jm] p. 17 -- You write that the algorithm "finally finds the shortest route between all cycle hire stations." By previous portions of the chapter, it is probably the shortest, and certainly close to the shortest, but cannot be guaranteed to actually be the shortest route. In this chapter, would it be possible to at least mention the existence of a bridge to ArcGIS Pro? Even though it isn't open source, ArcGIS Pro is very commonly used, and noting its R bridge would probably help increase the utilization of the R packages presented throughout the remainder of the book. [We have already mentioned the bridge to ArcGIS in a footnote previously, and have now expanded it.]
- [x] [jm]  p. 44 -- Your explanation of what a hyperparameter is would probably only make sense to people who already know what hyperparameters are, and thus don't need the explanation. Could you rephrase this to be more accessible? Something along the lines of "hyperparameters are the settings that control how the machine learning algorithm works, such as the number of trees in a random forest, rather than the parameters of a specific data-derived model" might be an improvement. [Thank you very much for your suggestion. Accordingly, we have rewritten the paragraph defining hyperparameters and added one further reference, see [d7930f7](https://github.com/geocompx/geocompr/commit/d7930f7f2acf59ef8ec1c582da8a07dfcfaedd56).]
- [x] [rl] p. 54 -- In the last paragraph, should "enable use to move" be "enable us to move"?
- [x] [rl] p. 78 -- I wouldn't change it, but still groan at the heading "future directions of travel"... No specific comments about the ecology chapter—maybe changing some of the color gradients to be more black/white friendly, but I recognize that I'm probably one of the few readers of this book who'd consider printing out the chapters with a black/white printer, so this may be an ignorable point.

Link to the PDF with the relevant page numbers is at the bottom of an email from Lara.

**Comment by Nowosad** [comment]
@Robinlovelace @jannes-m -- I gather all of the reviewer's new comments here.

**Comment by Robinlovelace** [comment]
Great work Jakub, will get on the case at some point during the festive period or a bit after perhaps :christmas_tree:

**Comment by Robinlovelace** [comment]
P. 54 issue fixed in commit above.

Re. p. 78 -- I wouldn't change it, but still groan at the heading "future directions of travel" no change needed.

**Comment by Robinlovelace** [comment]
Can you take a look at the final 2 points @jannes-m?

**Comment by jannes-m** [comment]
Yes, will do so the coming week.

**Comment by jannes-m** [comment]
Now, I have also addressed all comments raised by the reviewer.

**Comment by Robinlovelace** [comment]
Great work Jannes. Closing for now.
